### PR TITLE
Queue analytics pt 1

### PR DESCRIPTION
**Pull request by mmlumba**
Since there is some functionality that doesn't exist yet in Queue, only the following events have been created for this PR:
Queue table/home page:
* If a user clicks into an appeal's task details
* If a user clicks the reader documents link from Queue table

Queue tasks: 
* If a user clicks 'Open in Reader' from the task details view
* If a user clicks Appellant tab
* If a user clicks Appeal tab

This does not handle any of the cases that are opened in a new tab.

Connects #4343

### CODE REVIEW

File: client/app/queue/constants.js [hunk 1]
```diff
@@ -16,3 +16,22 @@ export const ACTIONS = {
...
+  VIEW_APPELLANT_INFO: 'view-appellant-info',
+  VIEW_APPEAL_INFO: 'view-appeal-info',
+  VIEW_QUEUE_TABLE: 'view-queue-table',
+  VIEW_QUEUE_TASK: 'view-queue-task',
+  QUEUE_TO_READER: 'queue-to-reader',
+  QUEUE_TO_HEARING_WORKSHEET: 'queue-to-hearing-worksheet'
+};
+
+export const INTERACTION_TYPES = {
+  NEW_WINDOW: 'new-window',
```
Comment (nickheiner-usds) [inline client/app/queue/constants.js]: We should not add values that are not used anywhere.

File: client/app/queue/QueueDetailView.jsx [hunk 2]
```diff
@@ -20,6 +20,10 @@ const subHeadStyling = css({
 });
 
 class QueueDetailView extends React.PureComponent {
+  componentDidMount() {
```
Comment (nickheiner-usds) [inline client/app/queue/QueueDetailView.jsx]: `QueueDetailView` should always mount as part of a `PageRoute`, which means that we get page level tracking for free. Is this call redundant with that?

File: client/app/queue/ReaderLink.jsx [hunk 3]
```diff
@@ -5,7 +5,14 @@ import _ from 'lodash';
 
 import Link from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/Link';
 
+import { TASK_ACTIONS } from './constants';
+
 class ReaderLink extends React.PureComponent {
+
+  readerLinkAnalytics = () => {
+    window.analyticsEvent(this.props.analyticsSource, TASK_ACTIONS.QUEUE_TO_READER, this.props.interactionType);
```
Comment (nickheiner-usds) [inline client/app/queue/ReaderLink.jsx]: The only `interactionType` we pass is `SAME_WINDOW`, but that's not always accurate. If I `command+click` on the link, for instance, it opens in a new tab, but we'll see a tracking event fire with `SAME_WINDOW`.

Determining if the user is opening the link in the same tab or a new tab is tricky. Let's get confirmation from the designers (like @laurjpeterson) that they think it's worth the effort before we do it. We can do the rest of the ticket in this PR.

File: client/app/queue/AppealDetail.jsx [hunk 4]
```diff
@@ -68,6 +68,10 @@ export default class AppealDetail extends React.PureComponent {
     return <BareList ListElementComponent="ul" items={listElements.map(getDetailField)} />;
   };
 
+  componentDidMount() {
+    window.analyticsEvent(CATEGORIES.QUEUE_TASK, TASK_ACTIONS.VIEW_APPEAL_INFO, INTERACTION_TYPES.SAME_WINDOW);
```
Comment (nickheiner-usds) [inline client/app/queue/AppealDetail.jsx]: There's no way to open either of the tabs in a new window, so I don't think that passing `SAME_WINDOW` or `NEW_WINDOW` makes sense here.

File: client/app/queue/ReaderLink.jsx (hunk 3 above)
Comment (mmlumba) [reply client/app/queue/ReaderLink.jsx]: I wrote in my PR description that I only covered events that were opened in the same window. I haven't covered tracking events that opened in a new window with this PR.

Comment (nickheiner-usds) [reply client/app/queue/ReaderLink.jsx]: It's reasonable to not cover all the cases in a single PR. My concern was that we were actually sending incorrect data, by sending `SAME_WINDOW` when the content was actually opened in a new window.

File: client/app/queue/constants.js [hunk 5]
```diff
@@ -17,3 +17,21 @@ export const ACTIONS = {
...
+
+export const CATEGORIES = {
+  QUEUE_TABLE: 'Queue Table',
+  QUEUE_TASK: 'Queue Task'
+};
+
+export const TASK_ACTIONS = {
+  VIEW_APPELLANT_INFO: 'view-appellant-info',
+  VIEW_APPEAL_INFO: 'view-appeal-info',
+  VIEW_QUEUE_TABLE: 'view-queue-table',
```
Comment (nickheiner-usds) [inline client/app/queue/constants.js]: We should not be adding constants that are not used.

**Pull request merged**